api/unversioned/config/config.go

```diff
@@ -0,0 +1,168 @@
+package config
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/Azure/eraser/api/unversioned"
+	"github.com/Azure/eraser/version"
+	"k8s.io/apimachinery/pkg/api/resource"
+)
+
+var defaultScannerConfig = `
+cacheDir: /var/lib/trivy
+dbRepo: ghcr.io/aquasecurity/trivy-db
+deleteFailedImages: true
+deleteEOLImages: true
+vulnerabilities:
+  ignoreUnfixed: true
+  types:
+    - os
+    - library
+securityChecks: # need to be documented; determined by trivy, not us
+  - vuln
+severities:
+  - CRITICAL
+  - HIGH
+  - MEDIUM
+  - LOW
+`
+
+type Manager struct {
+	mtx sync.Mutex
+	cfg *unversioned.EraserConfig
+}
+
+func (m *Manager) Read() (unversioned.EraserConfig, error) {
+	m.mtx.Lock()
+	defer m.mtx.Unlock()
+
+	if m.cfg == nil {
+		return unversioned.EraserConfig{}, fmt.Errorf("ConfigManager configuration is nil, aborting")
+	}
+
+	cfg := *m.cfg
+	return cfg, nil
+}
+
+func (m *Manager) Update(newC *unversioned.EraserConfig) error {
+	m.mtx.Lock()
+	defer m.mtx.Unlock()
+
+	if m.cfg == nil {
+		return fmt.Errorf("ConfigManager configuration is nil, aborting")
+	}
+
+	if newC == nil {
+		return fmt.Errorf("new configuration is nil, aborting")
+	}
+
+	*m.cfg = *newC
+	return nil
+}
+
+func NewManager(cfg *unversioned.EraserConfig) *Manager {
+	return &Manager{
+		mtx: sync.Mutex{},
+		cfg: cfg,
+	}
+}
+
+const (
+	noDelay = unversioned.Duration(0)
+	oneDay  = unversioned.Duration(time.Hour * 24)
+)
+
+func Default() *unversioned.EraserConfig {
+	return &unversioned.EraserConfig{
+		Manager: unversioned.ManagerConfig{
+			Runtime:      "containerd",
+			OTLPEndpoint: "",
+			LogLevel:     "info",
+			Scheduling: unversioned.ScheduleConfig{
+				RepeatInterval:   unversioned.Duration(oneDay),
+				BeginImmediately: true,
+			},
+			Profile: unversioned.ProfileConfig{
+				Enabled: false,
+				Port:    6060,
+			},
+			ImageJob: unversioned.ImageJobConfig{
+				SuccessRatio: 1.0,
+				Cleanup: unversioned.ImageJobCleanupConfig{
+					DelayOnSuccess: noDelay,
+					DelayOnFailure: oneDay,
+				},
+			},
+			PullSecrets: []string{},
+			NodeFilter: unversioned.NodeFilterConfig{
+				Type: "exclude",
+				Selectors: []string{
+					"eraser.sh/cleanup.filter",
+				},
+			},
+		},
+		Components: unversioned.Components{
+			Collector: unversioned.OptionalContainerConfig{
+				Enabled: false,
+				ContainerConfig: unversioned.ContainerConfig{
+					Image: unversioned.RepoTag{
+						Repo: repo("collector"),
+						// Tag:  version.BuildVersion,
```

Comment:
is the comment intentional?

Reply:
No. This was from when my brain go confused about overlaying on the defaults. It's fine to just have `version.BuildVersion` there.

Reply:
Fixed